src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,76 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static Random random = new Random();
+    private static ColorSupplier colorSupplier = new ColorSupplier();
+    private static final int NUMBER_OF_FIGURES = 5;
+    private static final int MAX_RANDOM_INT_VALUE = 100;
+
+    public Figure getRandomFigure() {
+        int randomIndex = random.nextInt(NUMBER_OF_FIGURES);
+
+        switch (randomIndex) {
+            case 0:
+                return getRandomCircle();
+            case 1:
+                return getRandomSquare();
+            case 2:
+                return getRandomRectangle();
+            case 3:
+                return getRandomIsoscelesTrapezoid();
+            case 4:
+                return getRandomRightTriangle();
+            default:
+                return getDefaultFigure();
+        }
+    }
+
+    private static Figure getRandomCircle() {
```

Comment:
All private methods should be below public methods
Methods should not be static, to fix for all
```suggestion
    private Figure getRandomCircle() {
```